tests/functional/Snowflake/Exporter/ExportTest.php

```diff
@@ -0,0 +1,238 @@
+
+        $files = $this->listFiles($this->getExportDir());
+        self::assertNotNull($files);
+        $actual = $this->getCsvFileFromStorage($files);
+        $expected = new CsvFile(
+            self::DATA_DIR . 'with-ts.csv',
+            CsvOptions::DEFAULT_DELIMITER,
+            CsvOptions::DEFAULT_ENCLOSURE,
+            CsvOptions::DEFAULT_ESCAPED_BY,
+            1 // skip header
+        );
+        $this->assertCsvFilesSame($expected, $actual);
+
+        $files = $this->getFileNames($this->getExportDir(), false);
+        $this->assertContains($this->getExportDir() . '/ts_testmanifest', array_values($files));
+    }
+
+    public function assertCsvFilesSame(CsvFile $expected, CsvFile $actual): void
+    {
+        $this->assertArrayEqualsSorted(
+            iterator_to_array($expected),
+            iterator_to_array($actual),
+            0,
+            'Csv files are not same'
+        );
+    }
+
+    public function testExportSimpleWithQuery(): void
+    {
+        $this->initTable(self::TABLE_ACCOUNTS_3);
+        // import
+        $file = new CsvFile(self::DATA_DIR . 'tw_accounts.csv');
+        $source = $this->getSourceInstance('tw_accounts.csv', $file->getHeader());
+        $destination = new Storage\Snowflake\Table(
+            $this->getDestinationSchemaName(),
+            self::TABLE_ACCOUNTS_3
+        );
+
+        $this->importTable($source, $destination);
+
+        // export
+        // query needed otherwise timestamp is downloaded
+        $query = sprintf(
+            'SELECT %s FROM %s',
+            ColumnsHelper::getColumnsString($file->getHeader()),
+            $destination->getQuotedTableWithScheme()
+        );
+        $source = new Storage\Snowflake\SelectSource($query);
+        $options = new ExportOptions();
+        $destination = $this->getDestinationInstance($this->getExportDir() . '/tw_test');
+
+        $result = (new Exporter($this->connection))->exportTable(
+            $source,
+            $destination,
+            $options
+        );
+
+        $this->assertCount(1, $result);
+        /** @var array<mixed> $slice */
+        $slice = reset($result);
+
+        $this->assertArrayHasKey('FILE_NAME', $slice);
+        $this->assertArrayHasKey('FILE_SIZE', $slice);
+        $this->assertArrayHasKey('ROW_COUNT', $slice);
+
+        $this->assertSame('tw_test_0_0_0.csv', $slice['FILE_NAME']);
+        $this->assertNotEmpty($slice['FILE_SIZE']);
+        $this->assertSame(3, (int) $slice['ROW_COUNT']);
+
+        $files = $this->listFiles($this->getExportDir());
+        self::assertNotNull($files);
+
+        $actual = $this->getCsvFileFromStorage($files);
+        $expected = new CsvFile(
+            self::DATA_DIR . 'tw_accounts.csv',
+            CsvOptions::DEFAULT_DELIMITER,
+            CsvOptions::DEFAULT_ENCLOSURE,
+            CsvOptions::DEFAULT_ESCAPED_BY,
+            1 // skip header
+        );
+        $this->assertCsvFilesSame($expected, $actual);
```

Comment:
Tohle netestuje, že se to vyexportovalo správně tou query, ale jen že soubor v S3 === soubor, co ti to poskládalo lokálně. Tj. vlastně, že se to správně staihlo, ne?

Reply:
Aha, ono to má níž unit test, to je tak netradiční, že mě to zaskočilo

Reply:
No, ten test testuje něco jako

> pokud exportuju query SELECT 1, id FROM someTable, tak na výstupu je CSV, které má 2 sloupce i když tabulka má 3 (vzala se v potaz ta query)

Ale tenhle assert testuje 

> soubor, který SNFLK uložil na S3 jako výsledek query je stejný jako ten, který mi to stahne ze sliců a složí na disku

Jakože - TDD pingpong by byl, že změním tu query a přidám ji další sloupec a ono to nespadne.

Reply:
Jo, ne, už to vidím. Je to OK. Tam se to tahá z disku těch testů a s tím se to porovnává. Nevím... nějaky brainfart